samples/mixed/provider/src/vehicle.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+const MAX_HVAC_TEMPERATURE: u32 = 100;
+const MIN_HVAC_TEMPERATURE: u32 = 65;
+
+#[derive(Debug, Default)]
+pub struct Vehicle {
+    pub ambient_air_temperature: u32,
+    pub is_air_conditioning_active: bool,
+    pub ui_message: String,
+    pub hybrid_battery_remaining: f32,
+}
+
+impl Vehicle {
+    pub fn new() -> Self {
+        Self {
+            ambient_air_temperature: 75,
+            is_air_conditioning_active: false,
+            ui_message: String::from(""),
+            hybrid_battery_remaining: 100.0,
+        }
+    }
+
+    pub fn execute_epoch(&mut self) {
+        // A/C will not be active without power.
+        if self.hybrid_battery_remaining == 0.0 {
+            self.is_air_conditioning_active = false;
+        }
+
+        // Adjust the ambient air temperature based on whether the A/C is being used.
+        if self.is_air_conditioning_active {
+            if self.ambient_air_temperature > MIN_HVAC_TEMPERATURE {
+                self.ambient_air_temperature -= 1;
+            }
+        } else if self.ambient_air_temperature < MAX_HVAC_TEMPERATURE {
```

Comment:
this looks like it should be nested in the above statement, otherwise the temp will still change even when the ac is inactive

Reply:
I don't like this either, but this was the condensed version that the static code analyzer insisted that I use.

Reply:
it's not that I don't like the formatting, it looks like the logic is incorrect. I'm suggesting the following instead:

```
if self.is_air_conditioning_active {
    if self.ambient_air_temperature > MIN_HVAC_TEMPERATURE {
        self.ambient_air_temperature -= 1;
    } else if self.ambient_air_temperature < MAX_HVAC_TEMPERATURE {
        self.ambient_air_temperature += 1;
    }
}
```

Reply:
or since this the intended behavior, add a comment explaining that the temperature should go up because the AC is off